src/main/java/servlets/SearchServlet.java

```diff
@@ -52,16 +59,17 @@
       "No Authentication: User must be logged in to search receipts.";
 
   private final DatastoreService datastore;
-  private final UserService userService;
+  private final UserService userService = UserServiceFactory.getUserService();
+
+  private PreparedQuery preparedQuery;
+  private QueryInformation queryInformation;
```

Comment:
Yeah QueryInformation can easily be made local but PreparedQuery is needed for pagination so it knows which query to apply the cursor to: https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Cursor. 

I would need to somehow send that information to the client and back like I did with the cursor, but I think it would take a lot of restructuring :/ I can try to do that

Reply:
When the user clicks on next or previous, can you just pass in all of the input parameters again used in `createQueryInformation()` to recreate the PreparedQuery?